src/Syntax/Resolve.hs

```diff
@@ -293,11 +298,21 @@ reExpr r@(Ascription e t a) = Ascription
                           <$> reExpr e
                           <*> reType r t
                           <*> pure a
-reExpr (Record fs a) = Record <$> traverse (traverse reExpr) fs <*> pure a
-reExpr (RecordExt e fs a) = RecordExt
-                        <$> reExpr e
-                        <*> traverse (traverse reExpr) fs
-                        <*> pure a
+reExpr e@(Record fs a) = do
+  let ls = map fst fs
+      nub = Set.toList . Set.fromList
+  case ls \\ nub ls of
+    l:_ -> throwError $ NonLinearRecord e l
```

Comment:
Can we do `tell (map (NonLinearRecord e) (nub (ll \\ nub ls)))` instead? This way we can report multiple errors.

Reply:
Hrmr, is there a reason for defining a custom `nub`? I assume it's more efficient, but as `n` is probably tiny, I don't know how much difference it'd make.

There's a bit of me which would argue something like `mapMaybe (listToMaybe . tail) . group . sort`, though maybe that's a overboard.